isis/src/base/objs/SpicePosition/SpicePosition.cpp

```diff
@@ -464,20 +484,28 @@ namespace Isis {
         j2000Coord.push_back((double)rec[0]);
         j2000Coord.push_back((double)rec[1]);
         j2000Coord.push_back((double)rec[2]);
+
         int inext = 3;
 
-        p_cache.push_back(j2000Coord);
         if (p_hasVelocity) {
           std::vector<double> j2000Velocity;
           j2000Velocity.push_back((double)rec[3]);
           j2000Velocity.push_back((double)rec[4]);
           j2000Velocity.push_back((double)rec[5]);
           inext = 6;
 
-          p_cacheVelocity.push_back(j2000Velocity);
+          stateCache.push_back(ale::State(j2000Coord, j2000Velocity));
+        }
+        else {
+          stateCache.push_back(ale::Vec3d(j2000Coord));
```

Comment:
How does this work if stateCache is a vector of `ale::State` objects?

Reply:
Kristin found this one. The States vector constructor requires 6 elements or it errors. Since, this vector is just position we need to convert it into a position vector and then a state for it to work correctly.